steps/expression_processor.go

```diff
@@ -52,10 +97,10 @@ func (p *ExpressionProcessor) AddExpression(expressionString string) error {
 }
 
 func (p *ExpressionProcessor) Sample(sample *bitflow.Sample, header *bitflow.Header) error {
-	if res, err := p.evaluate(sample, header); err != nil {
+	if outSample, outHeader, err := p.evaluate(sample, header); err != nil {
 		return err
-	} else if res {
-		return p.NoopProcessor.Sample(sample, header)
+	} else if sample != nil || header != nil {
```

Comment:
Turn the `||` to `&&`, because both values are necessary. We know that both are present, as long as one is present, but this is not obvious when reading just this line.

Reply:
Done